x-pack/plugins/security_solution/public/common/containers/source/index.tsx

```diff
@@ -85,96 +72,76 @@ export const getBrowserFields = memoizeOne(
       : {}
 );
 
-export const WithSource = React.memo<WithSourceProps>(({ children, indexToAdd, sourceId }) => {
+export const indicesExistOrDataTemporarilyUnavailable = (
+  indicesExist: boolean | null | undefined
+) => indicesExist || isUndefined(indicesExist);
+
+export const useWithSource = (sourceId = 'default', indexToAdd?: string[] | null) => {
   const [configIndex] = useUiSetting$<string[]>(DEFAULT_INDEX_KEY);
+  const [loading, updateLoading] = useState(false);
```

Comment:
something that we learned, it is better to create a complex state object than multiple states. With only one state, we will avoid a lot of re-render. We discover that in the detection development and we refactor most of api hooks to manage with one state.